### ISSUE TITLE
Refactor QC module

### ISSUE THREAD

**Issue by jeromekelleher** [popsim-consortium/stdpopsim#304]
The QC module currently uses subclasses of Model to define the relevant demographic models. This is quite boilerplatey. Refactor to use something similar to the function approach in #275.

We should wait on the model ID scheme being finalised first in #122 

Not necessary for shipping 0.1.0 since it's not part of the external API or behaviour.
